packages/dds/tree/src/test/shared-tree/sharedTreeFuzzTests.ts

```diff
@@ -0,0 +1,263 @@
+	initializeTestTree(provider.trees[0], initialTreeState, testSchema);
+	await provider.ensureSynchronized();
+
+	const initialState: FuzzTestState = {
+		random,
+		testTreeProvider: provider,
+		numberOfEdits: 0,
+	};
+	await initialState.testTreeProvider.ensureSynchronized();
+
+	const finalState = await performFuzzActionsBase(
+		generator,
+		{
+			edit: async (state, operation) => {
+				const { index, contents } = operation;
+				const tree = state.testTreeProvider.trees[index];
+				applyFuzzChange(tree, contents, TransactionResult.Apply);
+				return state;
+			},
+			synchronize: async (state) => {
+				const { testTreeProvider } = state;
+				await testTreeProvider.ensureSynchronized();
+				checkTreesAreSynchronized(testTreeProvider);
+				return state;
+			},
+		},
+		initialState,
+		saveInfo,
+	);
+	await finalState.testTreeProvider.ensureSynchronized();
+	return finalState;
+}
+
+export async function performFuzzActionsAbort(
+	generator: AsyncGenerator<Operation, FuzzTestState>,
+	seed: number,
+	saveInfo?: { saveAt?: number; saveOnFailure: boolean; filepath: string },
+): Promise<FuzzTestState> {
+	const random = makeRandom(seed);
+	const provider = await TestTreeProvider.create(1, SummarizeType.onDemand);
+	const tree = provider.trees[0];
+
+	initializeTestTree(provider.trees[0], initialTreeState, testSchema);
+	validateTree(provider.trees[0], [initialTreeState]);
+
+	// building the anchor for anchor stability test
+	const cursor = tree.forest.allocateCursor();
+	moveToDetachedField(tree.forest, cursor);
+	cursor.enterNode(0);
+	cursor.getPath();
+	cursor.firstField();
+	cursor.getFieldKey();
+	cursor.enterNode(1);
+	const firstAnchor = cursor.buildAnchor();
+	cursor.free();
+
+	const initialState: FuzzTestState = {
+		random,
+		testTreeProvider: provider,
+		numberOfEdits: 0,
+	};
+	const finalState = await performFuzzActionsBase(
+		generator,
+		{
+			edit: async (state, operation) => {
+				const { index, contents } = operation;
+				applyFuzzChange(tree, contents, TransactionResult.Abort);
+				return state;
+			},
+			synchronize: async (state) => {
+				return state;
+			},
+		},
+		initialState,
+		saveInfo,
+	);
+	validateTree(provider.trees[0], [initialTreeState]);
+
+	// validate anchor
+	const expectedPath: UpPath = {
+		parent: {
+			parent: undefined,
+			parentIndex: 0,
+			parentField: rootFieldKeySymbol,
+		},
+		parentField: brand("foo"),
+		parentIndex: 1,
+	};
+	const anchorPath = tree.locate(firstAnchor);
+	assert(compareUpPaths(expectedPath, anchorPath));
+	return finalState;
+}
+
+export function checkTreesAreSynchronized(provider: ITestTreeProvider) {
+	const tree0 = provider.trees[0];
+	for (let i = 1; i < 4; i++) {
+		const readCursor = provider.trees[i].forest.allocateCursor();
+		moveToDetachedField(provider.trees[i].forest, readCursor);
+		const currentTree = mapCursorField(readCursor, jsonableTreeFromCursor);
+		readCursor.free();
+		validateTree(tree0, currentTree);
```

Comment:
The current code converts both trees at each iteration of the loop.
You should convert tree0 once outside the loop and then call `validateTree(provider.trees[i], tree0Jsonable)` for each `i`.